static/js/lib/learning_resources.js

```diff
@@ -27,24 +34,131 @@ export const availabilityLabel = (availability: ?string) => {
   }
 }
 
+export const parseDateFilter = (filter: string) => {
+  const format = /(now)(\+)?(\d+)?([Md])?/
+  const match = format.exec(filter)
+  if (match) {
+    let dt = moment()
+    if (match[3] && match[4]) {
+      dt = dt.add(match[3], match[4] === "d" ? "days" : "months")
+    }
+    return dt
+  }
+}
+
+export const inDateRanges = (run: CourseRun, availabilities: Array<string>) => {
+  let inRange = false
+  availabilities.forEach(availability => {
+    if (AVAILABILITY_MAPPING[availability]) {
+      const from = parseDateFilter(
+        AVAILABILITY_MAPPING[availability].filter.from
+      )
+      const to = parseDateFilter(AVAILABILITY_MAPPING[availability].filter.to)
+      const startDate = run.best_start_date
+        ? moment(run.best_start_date, dateFormat)
+        : null
+      if (
+        ((isNil(startDate) && availability === AVAILABLE_NOW) ||
+          isNil(from) ||
+          // $FlowFixMe: okay to compare moments
+          startDate >= from) &&
+        ((isNil(startDate) && availability === AVAILABLE_NOW) ||
+          isNil(to) ||
+          // $FlowFixMe: okay to compare moments
+          startDate <= to)
+      ) {
+        inRange = true
+      }
+    }
+  })
+  return inRange
+}
```

Comment:
general question about this function, is the idea to return `true` if the `courseRun` is in 'range' for any of the availabilities supplied? because I think, based on the way it is now if there were an availability which wasn't "in range" after one which was we'd return `false` in a case where we want to return `true`.

Reply:
yes, if the user selects "Available Now" and "In the next week" for example, a run should be returned if it's dates fall within any of those date ranges specified.  `inRange` is set to `false` initially, then all the user-selected availabilities will be iterated over, and if any one of them is a match, it will be set to `true`.  Only if it doesn't find any match for any of the availabilities will it still return `false`.